dashboard/app/models/script.rb

```diff
@@ -192,6 +192,23 @@ def self.valid_scripts(user)
     end
   end
 
+  # Get script and course info for the current user, formatted for passing to
+  # the client for new teacher dashboard UI.
+  # @param [User] user
+  # @return [AssignableInfo[]]
+  def self.valid_assignable_info(user)
+    valid_scripts(user).
+      map do |course_or_script|
+        info = ScriptConstants.assignable_info(course_or_script)
+        info[:name] = I18n.t("#{info[:name]}_name", default: info[:name])
+        info[:name] += " *" if course_or_script[:hidden]
+
+        info[:category] = I18n.t("#{info[:category]}_category_name", default: info[:category])
```

Comment:
It turns out this issue is more complicated than moving localizations - we're using different course names in the assignment selector than we actually have in dashboard i18n, and there are still several missing, and we'd have to move category names too.

My first pass at addressing this will be making pegasus expose valid scripts in an API that we can call from dashboard too, so we use exactly the same assignments list.  Later we can deal with moving the assignment list/localizations as an independent change, probably after we are done removing the sections page.